flepimop/gempyor_pkg/src/gempyor/seeding.py

```diff
@@ -0,0 +1,155 @@
+    day_start_idx[1:] = np.cumsum(nb_seed_perday)
+    seeding_dict["day_start_idx"] = day_start_idx
+
+    return seeding_dict, seeding_amounts
+
+class Seeding(SimulationComponent):
+    def __init__(self, config: confuse.ConfigView):
+        self.seeding_config = config
+
+    def get_from_config(self, sim_id: int, setup) -> nb.typed.Dict:
+        method = "NoSeeding"
+        if self.seeding_config is not None and "method" in self.seeding_config.keys():
+            method = self.seeding_config["method"].as_str()
+
+        if method == "NegativeBinomialDistributed" or method == "PoissonDistributed":
+            seeding = pd.read_csv(
+                self.seeding_config["lambda_file"].as_str(),
+                converters={"subpop": lambda x: str(x)},
+                parse_dates=["date"],
+                skipinitialspace=True,
+            )
+            dupes = seeding[seeding.duplicated(["subpop", "date"])].index + 1
+            if not dupes.empty:
+                raise ValueError(f"Repeated subpop-date in rows {dupes.tolist()} of seeding::lambda_file.")
+        elif method == "FolderDraw":
+            seeding = pd.read_csv(
+                setup.get_input_filename(
+                    ftype=setup.seeding_config["seeding_file_type"].get(),
+                    sim_id=sim_id,
+                    extension_override="csv",
+                ),
+                converters={"subpop": lambda x: str(x)},
+                parse_dates=["date"],
+                skipinitialspace=True,
+            )
+        elif method == "FromFile":
+            seeding = pd.read_csv(
+                self.seeding_config["seeding_file"].get(),
+                converters={"subpop": lambda x: str(x)},
+                parse_dates=["date"],
+                skipinitialspace=True,
+            )
+        elif method == "NoSeeding":
+            seeding = pd.DataFrame(columns=["date", "subpop"])
+            return _DataFrame2NumbaDict(df=seeding, amounts=[], setup=setup)
+        else:
+            raise NotImplementedError(f"unknown seeding method [got: {method}]")
+
+        # Sorting by date is very important here for the seeding format necessary !!!!
+        # print(seeding.shape)
+        seeding = seeding.sort_values(by="date", axis="index").reset_index()
+        # print(seeding)
+        mask = (seeding["date"].dt.date > setup.ti) & (seeding["date"].dt.date <= setup.tf)
+        seeding = seeding.loc[mask].reset_index()
+        # print(seeding.shape)
+        # print(seeding)
+
+        # TODO: print.
+
+        amounts = np.zeros(len(seeding))
+        if method == "PoissonDistributed":
+            amounts = np.random.poisson(seeding["amount"])
+        elif method == "NegativeBinomialDistributed":
+            raise ValueError("Seeding method 'NegativeBinomialDistributed' is not supported by flepiMoP anymore.")
+            amounts = np.random.negative_binomial(n=5, p=5 / (seeding["amount"] + 5))
```

Comment:
Is this line needed after raising a ValueError?

Reply:
It looks like this one should be removed. @jcblemai  Should I move this to somewhere else?

Reply:
yes, you can remove it (we let it here so someone could see how this method used to be done.